### ISSUE TITLE
pad

### ISSUE THREAD

**Issue by hlky** [hlky/AITemplate#4]
[torch.nn.functional.pad](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html)

Currently constant padding is accomplished with a combination of ops.full and ops.concatenate, this is not ideal.

pad operator should support 'constant', 'reflect', 'replicate' and 'circular', and fill value for 'constant'

**Comment by hlky** [comment]
Closed by [aa84832](https://github.com/hlky/AITemplate/commit/aa84832f916965d47d10c8459f13c403ec85ea83)

**Comment by hlky** [comment]
TODO: channel pad on 4 dim. low priority
TODO: 5 dim. low priority
